src/bundle/ApiPlatform/OpenApiFactory.php

```diff
@@ -104,8 +104,8 @@ private function processOperationResponses(Operation $operation): Operation
 
                 $exampleFilePath = $this->kernel->locateResource($responseContent['x-ibexa-example-file']);
                 $exampleFileContent = file_get_contents($exampleFilePath);
-
-                $newContent[$mediaType]['example'] = $exampleFileContent;
+                $isJson = 'json' === array_slice(explode('.', pathinfo($exampleFilePath, PATHINFO_FILENAME)), -1, 1)[0];
+                $newContent[$mediaType]['example'] = $isJson ? json_decode($exampleFileContent ?: '', true) : $exampleFileContent;
```

Comment:
one more thing I forgot about, please, add the `JSON_THROW_ON_ERROR` flag to `json_decode` so we don't have silent fails.

Reply:
```suggestion
                $newContent[$mediaType]['example'] = $isJson ? json_decode($exampleFileContent ?: '', true, 512, JSON_THROW_ON_ERROR) : $exampleFileContent;
```